metaphor/snowflake/auth.py

```
@@ -89,17 +77,16 @@ def connect(config: SnowflakeAuthConfig) -> snowflake.connector.SnowflakeConnect
             encryption_algorithm=serialization.NoEncryption(),
         )
 
-        return snowflake.connector.connect(
-            account=config.account,
-            user=config.user,
-            private_key=pkb,
-            database=config.default_database,
-            application=METAPHOR_DATA_SNOWFLAKE_CONNECTOR,
-            session_parameters={
-                "QUERY_TAG": config.query_tag,
-            },
-        )
-
-    raise ValueError(
-        "Invalid Snowflake configuration, please set either password or private key"
+    return snowflake.connector.connect(
+        account=config.account,
+        user=config.user,
+        password=config.password,
+        private_key=pkb,
```

I just thought of an edge case: if a user provided both password and key, what will happen? Maybe we should add this to the validator too.

---

Good point. Internally it must prefer one over the other but we should fail loudly in that case. Will create a follow-up PR.

---

https://github.com/MetaphorData/connectors/pull/296